packages/block-editor/src/utils/block-bindings.js

```diff
@@ -1,18 +1,22 @@
 /**
  * WordPress dependencies
  */
-import { useDispatch, useSelect } from '@wordpress/data';
+import { useDispatch, useRegistry } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { store as blockEditorStore } from '../store';
 import { useBlockEditContext } from '../components/block-edit';
 
+function isObjectEmpty( object ) {
+	return ! object || Object.keys( object ).length === 0;
+}
+
 export function useBlockBindingsUtils() {
 	const { clientId } = useBlockEditContext();
 	const { updateBlockAttributes } = useDispatch( blockEditorStore );
-	const { getBlockAttributes } = useSelect( blockEditorStore );
+	const { getBlockAttributes } = useRegistry().select( blockEditorStore );
```

Comment:
Is there a benefit to doing it for `dispatch` as well?

Reply:
`dispatch` doesn't establish event listeners, so it shouldn't make any difference. @jsnajdr, can you chime in?

Reply:
`useDispatch` is implemented as `useRegistry().dispatch( store )`, so changing the code has no effect. `dispatch` indeed doesn't establish any event listeners.

Even this change for `useSelect` is basically a noop, because `useSelect( store )` is a special form that returns `useRegistry().select( store )`. Only the callback form (`useSelect( ( select ) => ... )`) establishes listeners.

All in all, the `useBlockBindingsUtils` doesn't establish any local state or listeners and will never trigger rerenders. It merely returns two callbacks that will read from the store when called, and will see the value that is in the store at that moment.

Reply:
Ah, cool, so no change is necessary. That wasn't clear enough when reading the post on Make Core.